common-pg/src/main/scala/com/socrata/pg/store/SqlIndexCleanup.scala

```diff
@@ -37,8 +38,13 @@ class SqlIndexCleanup() extends IndexCleanup {
               true
             } catch {
               case ex: SQLException =>
-                conn.clearWarnings()
-                log.error(s"sql error dropping index $name ${ex.toString}")
+                QueryRuntimeError(ex) match {
+                  case None =>
+                    log.error(s"SQL Exception drop index $name (${ex.getSQLState})", ex)
+                  case Some(e) =>
+                    log.warn(s"SQL Exception drop index $name ${e.getClass.getSimpleName} (${ex.getSQLState})")
+                }
+                conn.clearWarnings() // reset connection on the basis of autoCommit being true
```

Comment:
You say the clearWarnings only seems to work with autoCommit. I see that it has been left here but removed from indexManager. Is that because one path is associated with autoCommit?

Reply:
Yes, that's the reason we can keep using it in index drop.